Tools/src/FormatCheckTool/FormatCheckTool.cs

```diff
@@ -0,0 +1,234 @@
+                    {
+                        flag = $"JSON File is Empty, {PleaseCheckStr}";
+                    }
+                }
+                catch (JsonReaderException jre)
+                {
+                    flag = $"JSON Reader Exception! {PleaseCheckStr}";
+                }
+                catch (JsonSerializationException jse)
+                {
+                    flag = $"JSON SerialzationException! {PleaseCheckStr}";
+                }
+                catch (Exception e)
+                {
+                    flag = $"Other File Error, {PleaseCheckStr}";
+                }
+            }
+
+            return new Tuple<bool, List<TestModel>>(ConsoleStageResult("JSON Valid", flag), specs);
+        }
+
+        private static bool CheckBasicContent(List<TestModel> data)
+        {
+            List<string> res = new List<string> { };
+            foreach (TestModel spec in data)
+            {
+                res.Add(CheckOneBasicContentSpec(spec));
+            }
+
+            return ConsoleStageResult("Content Valid", string.Join("|", res), true);
+        }
+
+        private static bool CheckModelResult(List<TestModel> data)
+        {
+            string flag = string.Empty;
+            return ConsoleStageResult("Model Result", flag);
+        }
+
+        private static string CheckOneBasicContentSpec(TestModel spec)
+        {
+            string flag = string.Empty;
+            if (spec.Input.Equals(string.Empty))
+            {
+                flag = "No Input";
+            }
+            else
+            {
+                foreach (object result in spec.Results)
+                {
+                    string resultStr = CheckOneBasicContentSpecResult(spec, result);
+                    if (!resultStr.Equals(string.Empty))
+                    {
+                        flag = resultStr;
+                        break;
+                    }
+                }
+            }
+
+            return flag;
+        }
+
+        private static string CheckOneBasicContentSpecResult(TestModel spec, object result)
+        {
+            var serializeResult = JsonConvert.SerializeObject(result);
+            var dynamicResult = JsonConvert.DeserializeObject<dynamic>(serializeResult);
+            if (dynamicResult["Text"] == null || dynamicResult["Start"] == null || (dynamicResult["End"] == null && dynamicResult["Length"] == null))
+            {
+                return "Result Parameter Loss";
+            }
+
+            int start = dynamicResult["Start"];
+            int end = dynamicResult["End"] != null ? dynamicResult["End"] : dynamicResult["Length"] + start - 1;
+            var startEndStr = spec.Input.Substring(start, end - start + 1);
+            if (dynamicResult["Text"] != startEndStr)
+            {
+                return "Index Error";
+            }
+
+            return string.Empty;
+        }
+
+        private static bool ConsoleStageResult(string stage, string flag, bool isList = false)
```

Comment:
This seems a bit complicated to me. What exactly is happening here that can't be represented in a list of log lines?